events/wc_sacrifice_events.txt

```diff
@@ -0,0 +1,305 @@
+			trait = cruel
+		}
+		modifier = {
+			factor = 0.75
+			trait = kind
+		}
+	}
+
+	immediate = {
+		FROM = { save_event_target_as = target_sacrifice_location }
+		save_event_target_as = target_sacrificer_unit_owner
+
+		if = { # Get Offmap Old Gods Currency if worship Old Gods
+			limit = { publicly_worship_one_of_old_gods_trigger = yes }
+
+			random_list = {
+				10 = {
+					if = {
+						limit = { any_owned_bloodline = { has_bloodline_flag = bloodline_insanity_gain } }
+						set_variable = { which = local_received_offmap_old_gods_currency_insanity_from_province_population_sacrifice_variable value = 900 }
+					}
+					else = { set_variable = { which = local_received_offmap_old_gods_currency_insanity_from_province_population_sacrifice_variable value = 600 } }
+				}
+				20 = {
+					if = {
+						limit = { any_owned_bloodline = { has_bloodline_flag = bloodline_insanity_gain } }
+						set_variable = { which = local_received_offmap_old_gods_currency_insanity_from_province_population_sacrifice_variable value = 600 }
+					}
+					else = { set_variable = { which = local_received_offmap_old_gods_currency_insanity_from_province_population_sacrifice_variable value = 400 } }
+				}
+				30 = {
+					trigger = { FROM = { is_richer_holding = no } }
+
+					if = {
+						limit = { any_owned_bloodline = { has_bloodline_flag = bloodline_insanity_gain } }
+						set_variable = { which = local_received_offmap_old_gods_currency_insanity_from_province_population_sacrifice_variable value = 300 }
+					}
+					else = { set_variable = { which = local_received_offmap_old_gods_currency_insanity_from_province_population_sacrifice_variable value = 200 } }
+				}
+				40 = {
+					trigger = { FROM = { is_rich_holding = no } }
+
+					if = {
+						limit = { any_owned_bloodline = { has_bloodline_flag = bloodline_insanity_gain } }
+						set_variable = { which = local_received_offmap_old_gods_currency_insanity_from_province_population_sacrifice_variable value = 120 }
+					}
+					else = { set_variable = { which = local_received_offmap_old_gods_currency_insanity_from_province_population_sacrifice_variable value = 80 } }
+				}
+			}
+		}
+		else_if = { # Or get sacrifice points
+			limit = { religion_have_ritual_sacrifice_trigger = yes }
+
+			bloodthirsty_tiered_traits_effect = yes
+		}
+
+		piety = 25
+
+		FROM = { # Sacrifice province population
+			add_holding_modifier = { name = inhabitants_sacrificed years = 5 }
+			if = {
+				limit = { settlement_can_be_destroyed_trigger = yes }
+				destroy_random_building = THIS
+				location = {
+					if = {
+						limit = { has_wonder = yes }
+						random = {
+							chance = 30
+							wealth = 100
+							wonder = { set_wonder_damaged = yes }
+						}
+					}
+				}
+			}
+
+			if = { # Pick character to sacrifice
+				limit = { is_capital = yes }
+				owner = {
+					random_courtier = {
+						limit = {
+							is_ruler = no
+							can_be_sacrificed_trigger = yes
+						}
+						random = {
+							chance = 25
```

Comment:
Strange code. It's randomness in randomness thus it's super rare. Have you thought about using [preferred limits](https://ck2.paradoxwikis.com/Scripting#Preferred_Limit) or [score values](https://ck2.paradoxwikis.com/Scripting#Score_Value)?

Reply:
> It's randomness in randomness thus it's super rare.

Randomness in random event is intended.
Imo, it looks kinda weird if besieger always sacrifice one character.
Can change that and make so this event sacrifice 0-3 characters.

Reply:
Implemented, check it, pls